internal/db/remote/commit/commit.go

```diff
@@ -172,7 +173,62 @@ func run(p utils.Program, url string) error {
 		}
 	}
 
-	// 2. Create shadow db and run migrations.
+	timestamp := utils.GetCurrentTimestamp()
+
+	// 2. Special case if this is the first migration
+	{
+		localMigrations, err := os.ReadDir(filepath.Join("supabase", "migrations"))
+		if err != nil {
+			return err
+		}
+
+		if len(localMigrations) == 0 {
+			strings.Join(utils.InternalSchemas, "|")
+
+			// Use pg_dump instead of schema diff
+			out, err := utils.DockerRun(
+				ctx,
+				dbId,
+				&container.Config{
+					Image: utils.DbImage,
+					Env:   []string{"POSTGRES_PASSWORD=postgres"},
+					Entrypoint: []string{
+						"sh", "-c",
+						"pg_dump --schema-only --quote-all-identifier --exclude-schema '" + strings.Join(utils.InternalSchemas, "|") + `' --schema '*' --extension '*' --dbname '` + url + `' | sed 's/CREATE SCHEMA "public"/-- CREATE SCHEMA "public"/'`,
```

Comment:
Probably more maintainable to externalise these scripts so we can lint with shellcheck. Not necessarily in this PR.

Reply:
:+1: can convert all these in another PR